pkg/store/prometheus.go

```diff
@@ -312,6 +312,35 @@ func (p *PrometheusStore) LabelNames(ctx context.Context, r *storepb.LabelNamesR
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
 
+func deduplicateAndSortStringSlice(srtings []string) []string {
+	sort.Strings(srtings)
+	j := 0
+	for i := 1; i < len(srtings); i++ {
+		if srtings[j] == srtings[i] {
+			continue
+		}
+		j++
+		srtings[i], srtings[j] = srtings[j], srtings[i]
+	}
+	return srtings[:j+1]
+}
+
+// extendSortAndDeduplicateLabelValues adds values of matching label name from external labels.
+func (p *PrometheusStore) extendSortAndDeduplicateLabelValues(labelName string, labelValues []string) []string {
```

Comment:
Hm... why isn't this as easy as:
```
if requested_label is in externalLabel:
  return externallabel value
```

We overwrite with external label values always (we never honour the underlying ones), so there is no need for duplication etc.

Can we also double check the Prometheus behaviour on duplication in external labels itself? (same label name, 2 different values) - because I think it is forbidden.